core/modules/qdisp/JobQuery.cc

```diff
@@ -82,8 +82,8 @@ bool JobQuery::runJob() {
 
         auto qr = std::make_shared<QueryResource>(shared_from_this());
         std::lock_guard<std::recursive_mutex> lock(_rmutex);
-        if (_jobDescription->getAttemptCount() <= _getMaxAttempts()) {
-            bool okCount = _jobDescription->incrAttemptCount();
+        if (_jobDescription->getAttemptCount() < _getMaxAttempts()) {
```

Comment:
The condition check in the if statement relies upon a limit which is disconnected from a test made inside method JobDescription::incrAttemptCountScrubResults() relying upon this macro definition:

    MAX_JOB_ATTEMPTS